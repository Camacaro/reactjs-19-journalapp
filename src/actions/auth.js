import Swal from 'sweetalert2'

import { types } from "../types/types"
import {firebase, googleAuthProvider } from '../firebase/firebaseConfig'
import { startLoading, finishLoading } from "./ui"

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {

    dispatch( startLoading() );
    
    firebase.auth().signInWithEmailAndPassword( email, password )
      .then( userCredentials => {
        const { user } = userCredentials;

        dispatch( login(user.uid, user.displayName) );
        dispatch( finishLoading() );

      } )
      .catch( e => {
        console.log(e)
        dispatch( finishLoading() );

        Swal.fire('Error', e.message, 'error' );
      })

    
  }
}

export const startRegisterWithEmailPasswordName = ( email, password, name ) => {
  // Esto es gracias al thunk
  return (dispatch) => {

    firebase.auth().createUserWithEmailAndPassword(email, password)
      .then( async (userCredentials) => {
        const { user } = userCredentials;

        await user.updateProfile({ displayName: name })

        dispatch(
          login( user.uid, user.displayName )
        )
      })
      .catch( e => {
        console.log(e);
        Swal.fire('Error', e.message, 'error' );
      })
  }
}

export const startGoogleLogin = () => {
  return (dispatch) => {

    firebase.auth().signInWithPopup( googleAuthProvider )
      .then( userCredentials => {
        const { user } = userCredentials;
        // console.log(userCredentials)
        dispatch(
          login( user.uid, user.displayName )
        )
      })
  }
}


export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
      uid,
      displayName
    }
  }
)

export const startLogout = () => {
  // Async
  return async ( dispatch ) => {
    await firebase.auth().signOut();

    dispatch( logout() )
  }
}

export const logout = () => ({
  type: types.logout
})