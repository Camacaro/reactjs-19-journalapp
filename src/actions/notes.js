import { db } from "../firebase/firebaseConfig";
import { types } from "../types/types";


export const startNewNote = () => {
  // getState, es un segundo parametro que me permite obtener el estado a traves del thunk
  return async ( dispatch, getState ) => {

    const { uid } = getState().auth;
    
    const newNote = {
      title: '',
      body: '',
      date: new Date().getTime()
    }

    // collection where i want to save the note
    const docRef = await db.collection(`${ uid }/journal/notes`).add(newNote)

    console.log(docRef)

    dispatch( activeNote( docRef.id, newNote ) )
  }
}

export const activeNote = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note
  }
})