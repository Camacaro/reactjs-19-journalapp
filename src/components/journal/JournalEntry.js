import React from 'react'

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">

      <div 
        className="journal__entry-picture"
        style={{
          backgroundSize: 'cover',
          backgroundImage: 'url(https://www.hdnicewallpapers.com/Walls/Big/Cat/Beautiful_Cat_Imge_Download.jpg)'
        }}
      >
      </div>

      <div className="journal__entry-body">
        <p className="journal__entry-title">
          Un nuevo dia
        </p>
        <p className="journal__entry-content">
          Pariatur Lorem ipsum in non est aliquip Lorem do fugiat anim dolor officia esse nisi.
        </p>
      </div>

      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>28</h4>
      </div>

    </div>
  )
}
