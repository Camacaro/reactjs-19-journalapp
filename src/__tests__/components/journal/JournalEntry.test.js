import '@testing-library/jest-dom'
import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Pruebas <NoteScreen />', () => {

  const initState = {};

  // State de mi Redux en la app - lo que se ve en el Redux DevsTools
  let store = mockStore(initState);
  store.dispatch = jest.fn();

  const note = {
    id: 10,
    date: 0,
    title: 'lorem',
    body: 'Minim et aute ex exercitation deserunt qui aute cupidatat consequat eiusmod magna adipisicing.',
    url: 'https://algunlugar.com/foto.jpg'
  }

  const wrapper = mount( 
    <Provider store={ store }>
      <JournalEntry { ...note } /> 
    </Provider>
  )
  
  test('debe de mostrarse corectamente', () => {

    expect( wrapper ).toMatchSnapshot();
  })

  test('debe de activar la nota', () => {
    wrapper.find('.journal__entry').prop('onClick')();

    expect( store.dispatch ).toHaveBeenCalled()

    expect( store.dispatch ).toHaveBeenCalledWith(
      activeNote( note.id, { ...note } )
    )
  })
  
  
})