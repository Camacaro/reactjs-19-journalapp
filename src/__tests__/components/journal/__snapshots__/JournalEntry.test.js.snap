// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <NoteScreen /> debe de mostrarse corectamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://algunlugar.com/foto.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      lorem
    </p>
    <p
      className="journal__entry-content"
    >
      Minim et aute ex exercitation deserunt qui aute cupidatat consequat eiusmod magna adipisicing.
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
